docs/docs/sourcing-from-the-filesystem.md

```diff
@@ -2,7 +2,105 @@
 title: Sourcing from the Filesystem
 ---
 
-This is a stub. Help our community expand it.
+This guide will walk you through sourcing from the filesystem.
 
-Please use the [Gatsby Style Guide](/docs/gatsby-style-guide/) to ensure your
-pull request gets accepted.
+## Setup
+
+This guide assumes that you have a Gatsby project set up. If you need to set up a project, please reference the [Quick Start Guide](https://github.com/gatsbyjs/gatsby/tree/master/docs). In addition, it will be useful if you are familiar with Graph_i_QL, a tool that helps you structure your queries correctly.
+
+## Introducing Graph_i_QL
+
+Graph_i_QL is the GraphQL integrated development environment (IDE). It's a powerful (and all-around awesome) tool
+you'll use often while building Gatsby websites.
+
+You can access it when your site's development server is running—normally at
+<http://localhost:8000/___graphql>.
+
+<video controls="controls" autoplay="true" loop="true">
+  <source type="video/mp4" src="/graphiql-explore.mp4"></source>
+  <p>Your browser does not support the video element.</p>
+</video>
+
+Here you poke around the built-in `Site` "type" and see what fields are available
+on it—including the `siteMetadata` object you queried earlier. Try opening
+Graph_i_QL and play with your data! Press <kbd>Ctrl + Space</kbd> (or use <kbd>Shift + Space</kbd> as an alternate keyboard shortcut) to bring up
+the autocomplete window and <kbd>Ctrl + Enter</kbd> to run the GraphQL query. You'll be
+using Graph_i_QL a lot more through the remainder of the tutorial.
+
+## Sourcing plugins
+
+Data in Gatsby sites can come from anywhere: APIs, databases, CMSs,
+local files, etc.
+
+Source plugins fetch data from their source. E.g. the filesystem source plugin
+knows how to fetch data from the file system. The WordPress plugin knows how to
+fetch data from the WordPress API. We will focus on sourcing from the filesystem.
+
+First, we will add [`gatsby-source-filesystem`](/packages/gatsby-source-filesystem/) and
+explore how it works.
+
+Install the plugin at the root of the project:
+
+```sh
+npm install --save gatsby-source-filesystem
+```
+
+Then add it to your `gatsby-config.js`:
+
+```javascript{6-12}:title=gatsby-config.js
+module.exports = {
+  siteMetadata: {
+    title: `Pandas Eating Lots`,
+  },
+  plugins: [
+    {
+      resolve: `gatsby-source-filesystem`,
+      options: {
+        name: `src`,
+        path: `${__dirname}/src/`,
+      },
+    },
+    `gatsby-plugin-emotion`,
+    {
+      resolve: `gatsby-plugin-typography`,
+      options: {
+        pathToConfigModule: `src/utils/typography`,
+      },
+    },
+  ],
+}
+```
+
+Save that and restart the gatsby development server. Then open up Graph_i_QL
```

Comment:
We're not doing a step by step tutorial here -- I think what we could do here is

1. show an example file that gets sourced
2. show how to include the plugin (already done above)
3. show an example GraphiQL query (without explaining GraphiQL).

When we use a resource like GraphiQL here, we can link directly to where it's explained in other sections. For example, "Not familiar with GraphiQL? Check out this section of the tutorial that introduces it."